speeduino/sensors.ino

```diff
@@ -439,52 +692,77 @@ void readTPS(bool useFilter)
     else { currentStatus.CTPSActive = digitalRead(pinCTPS); } //Inverted mode (5v activates closed throttle position sensor)
   }
   else { currentStatus.CTPSActive = 0; }
-  TPS_time = micros();
+
+  
+  return adcState;
+}
+//Get the TPS change rate 
+void readTPSdot(){
+  int TPSrateOfChange;
+  static uint8_t TPSlast; //The previous TPS reading  
+  //note here that TPS read frequency is specially chosen 40Hz to get optimally fast and accurate tpsDOT calculation(to have whole numbers in integer multiplication)
+  TPSrateOfChange = (TPS_READ_FREQUENCY/20) * (currentStatus.TPS-TPSlast); //This is the % per second that the TPS has moved
```

Comment:
To me both formulas seem matematically equivalent. Only written differently. Also the `TPS_READ_FREQUENCY ` should resolve to constant 40. Will alter the constants.

Reply:
They are mathematically equivalent, but since we're working with integers, they're not computationally equivalent.
Let's say that FREQUENCY is 30 and (TPS-last) is 5 for the sake of this example :

Your way : 
(freq/20)*(current-last)
(30/20)*5
(1)*5
5

Vitor's way :
(freq*(current-last))/20
(30*5)/20
150/20
7

Still not the real answer (7.5), but it's the closest we can get with integers.

Reply:
Since we are discussing this, I have a suggestion to speed up the process by using consts too
`const int8_t tpsFreq = (TPS_READ_FREQUENCY * 16) / 20; //Scale time to correct shift count next
TPSrateOfChange = (tpsFreq * (currentStatus.TPS - TPSlast)) >> 4; //Note, the value 20 was changed to result return faster(just shifting)`

Reply:
@brunob45 @VitorBoss 
Current situation is that the exact computation actually performed is:
`TPSrateOfChange=(currentStatus.TPS - TPSlast) << 2;` (complier does that)
This is why I changed the reading frequency to 40Hz. If this is correct I would leave it at that.
defined TPS_READ_FREQUENCY can not really be changed by changing its defined value. Should I delete the define and hard code in the bit shift? As it can be misleading #define as it is really more there to indicate the current value, but can not be altered. Then would put the formula as it is as a comment?

Reply:
So, if I understood right, you changed the read frequency to 40 just to make your formula works?

Reply:
@VitorBoss , yes. I noticed there is integer math with fractions, and changed the frequency to avoid computation with fractions.